brave-kafka/src/main/java/smartthings/brave/kafka/producers/TracingProducerInterceptor.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Copyright 2016-2017 SmartThings
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package smartthings.brave.kafka.producers;
+
+import brave.Span;
+import brave.Tracer;
+import brave.propagation.TraceContext;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.SpanIdUtils;
+import com.github.kristofa.brave.SpanId;
+import com.google.protobuf.ByteString;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.config.ConfigException;
+import smartthings.brave.kafka.EnvelopeProtos;
+import zipkin.Constants;
+import zipkin.Endpoint;
+
+import java.util.Map;
+
+/**
+ * Uses {@link Tracer} to create one-way {@link brave.Span#flush()} span (With "cs" and "sr" only)
+ * @param <K>
+ */
+public class TracingProducerInterceptor<K> implements ProducerInterceptor<K, byte[]> {
+
+  private Tracer tracer;
+  private ClientTracer clientTracer;
+  private SpanNameProvider<K> nameProvider;
+  private final Endpoint kafkaEndpoint;
+
+
+  public TracingProducerInterceptor() {
+    kafkaEndpoint = Endpoint.builder().serviceName("Kafka").build();
+  }
+
+  @Override
+  public ProducerRecord<K, byte[]> onSend(ProducerRecord<K, byte[]> record) {
+    Span span;
+    if (clientTracer != null && SpanIdUtils.getMaybeParent(clientTracer) != null ) {
```

Comment:
This is great if it works.  I think you answered this but I just want to make sure that Kafka will definitely call this on the same thread that is handling the HTTP request.  If there is any chance that it doesn't we could end up with client spans attached to the wrong server span.